src/components/consul/selectors/DocumentSelector.js

```diff
@@ -60,6 +60,66 @@ export const DocumentSelector = ({ control, field, item }) => {
     setInfoAndErrorText(deleteArrayItem(infoAndErrorText, index));
   };
 
+  if (isVolunteer) {
+    return (
+      <>
+        <Input {...item} control={control} hidden name={name} value={JSON.stringify(value)} />
+        <RegularText smallest placeholder>
+          {infoText}
+        </RegularText>
+
+        <Button
+          title={buttonTitle}
+          invert
+          onPress={async () => {
+            const { name: title, size, uri, mimeType } = await selectDocument();
+
+            if (!uri) return;
+
+            /* the server does not support files more than 10MB in size. */
+            const errorText = size > 10485760 && texts.volunteer.imageGreater10MBError;
+
+            setDocumentsAttributes([...documentsAttributes, { uri, mimeType }]);
+
+            setInfoAndErrorText([
+              ...infoAndErrorText,
+              {
+                errorText,
+                infoText: `${title}`
+              }
+            ]);
```

Comment:
the `onPress` functions are nearly the same for volunteer or consul. please create a function that can be reused and handles both cases.

Reply:
Code repetition greatly reduced
https://github.com/ikuseiGmbH/smart-village-app-app/pull/606/commits/ba048ed71a501a766f49e70c8bc6dbc33e9c673f